src/base/rss/rss_autodownloader.cpp

```diff
@@ -375,10 +375,24 @@ void AutoDownloader::handleTorrentAdded(const QString &source)
     }
 }
 
-void AutoDownloader::handleAddTorrentFailed(const QString &source, [[maybe_unused]] const BitTorrent::AddTorrentError &error)
+void AutoDownloader::handleAddTorrentFailed(const QString &source, const BitTorrent::AddTorrentError &error)
 {
-    m_waitingJobs.remove(source);
-    // TODO: Re-schedule job here.
+    const auto job = m_waitingJobs.take(source);
+    if (!job)
+        return;
+
+    if (error.kind == BitTorrent::AddTorrentError::DuplicateTorrent)
```

Comment:
If you intent to add more error types to `BitTorrent::AddTorrentError` later then it would be better to use a `switch ()`.

Reply:
> If you intent to add more error types to `BitTorrent::AddTorrentError` later then it would be better to use a `switch ()`.

I prefer not to run ahead of the train in such matters.

Reply:
> I prefer not to run ahead of the train in such matters.

And it looks like I'm not alone in this:
[YAGNI](https://en.wikipedia.org/wiki/You_aren%27t_gonna_need_it)

Reply:
Whatever but just remember to use `switch ()` when the time comes.